test/saucelabs/helpers.py

```diff
@@ -21,16 +21,27 @@ def bCheckIfUrlHasChanged(driver):
         return 1
 
 
-def waitForUrlChange(driver, oldUrl, timeout=15):
+def waitForUrlChange(driver, pattern, find=True, timeout=10):
     t0 = time.time()
-    newUrl = oldUrl
-    while oldUrl == newUrl:
-        newUrl = driver.current_url
-        time.sleep(1)
-        ti = time.time()
-        tf = ti - t0
-        if tf > timeout:
-            raise Exception('%s has not changed in %.2f seconds' % (oldUrl, tf))
+    newUrl = driver.current_url
```

Comment:
Btw: this will always fail on IE9 because there's not push history support there which we use to update the URL of the addreaa bar.